compiler/rustc_lint/src/types.rs

```diff
@@ -672,6 +672,10 @@ crate fn ty_is_known_nonnull<'tcx>(cx: &LateContext<'tcx>, ty: Ty<'tcx>, mode: C
                 return true;
             }
 
+            if def.repr.hide_niche() {
```

Comment:
Could you add a comment here explaining why, for someone who may not already be aware when niches are hidden?

Reply:
Added a comment mentioning `no_niche` and `UnsafeCell`.